index.js

```diff
@@ -162,6 +164,58 @@ const inventoryCommand = (message) => {
   });
 };
 
+const lookUpCommand = (message) => {
+  const originalMessage = message.content;
+  const fuzzySearchKey = originalMessage.split(" ").splice(1).join(" ");
+  const fuzzySearchUrl =
+    "https://db.ygoprodeck.com/api/v7/cardinfo.php?&fname=";
+
+  console.log(fuzzySearchUrl + fuzzySearchKey);
+
+  axios
+    .get(fuzzySearchUrl + fuzzySearchKey)
+    .then((cardSearchResults) => {
+      const objectsReturnedBySearch = cardSearchResults.data.data;
+      const exactMatchObject = objectsReturnedBySearch.find(
+        (element) => element.name.toLowerCase() === fuzzySearchKey.toLowerCase()
+      );
+
+      const closestMatchObject = objectsReturnedBySearch.find((element) =>
+        element.name.toLowerCase().startsWith(fuzzySearchKey.toLowerCase())
+      );
+
+      const bestCardResult =
+        exactMatchObject || closestMatchObject || objectsReturnedBySearch[0];
+
+      const attachment = new Discord.MessageAttachment(
+        bestCardResult.card_images[0].image_url
+      );
+      message.reply(
+        "The top result for your search is: " + bestCardResult.name
+      );
+      message.channel.send(attachment);
+
+      const suggestedSearches = objectsReturnedBySearch
+        .filter((item) => item.name != bestCardResult.name)
+        .slice(0, 6)
+        .map((item, index) => "•  " + item.name);
+
+      if (suggestedSearches.length >= 1) {
```

Comment:
there will be suggested searches as long as `objectsReturnedBySearch` is more than 1. If you switch the conditional to be based on this instead, you can move `const suggestedSearches` under this conditional, and save some CPU cycles